buffer.go

```diff
@@ -1,35 +1,40 @@
 package fuse
 
-import "unsafe"
+import (
+	"fmt"
+	"unsafe"
+)
 
-// buffer provides a mechanism for constructing a message from
-// multiple segments.
+// buffer provides a mechanism to write out a message in segments. This is useful because
+// the fuse protocol means we have to assemble one contiguous byte array with several headers
+// and then data.
 type buffer []byte
 
-// alloc allocates size bytes and returns a pointer to the new
-// segment.
-func (w *buffer) alloc(size uintptr) unsafe.Pointer {
-	s := int(size)
+func newBuffer(buf *buffer, bytes []byte) {
```

Comment:
Not sure about naming, this is more like resetBuffer(). Add a comment in any case.